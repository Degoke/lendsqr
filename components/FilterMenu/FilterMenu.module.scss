.popover {
  bottom: 0;
}

.form {
  display: grid;
  grid-template-columns: 1fr;
  gap: 2rem;
  padding: 1rem;
  width: 300px;

  input,
  select {
    height: 40px;
    border: 1px solid #213f7d;
    border-radius: 8px;
  }

  button {
    height: 40px;
    border-radius: 8px;
  }
}
