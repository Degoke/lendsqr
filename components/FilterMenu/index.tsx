import Popover from '../Popover'
import styles from './FilterMenu.module.scss'

type Props = {
  isOpen: boolean
}

export default function FilterMenu({ isOpen }: Props) {
  return (
    <Popover isOpen={isOpen} className={styles.popover}>
      <form className={styles.form} name="filterForm">
        <select>
          <option>select</option>
        </select>
        <input type="" />
        <input type="" />
        <input type="" />
        <input type="" />
        <select>
          <option>select</option>
        </select>
        <div>
          <button>Reset</button>
          <button type="submit">Filter</button>
        </div>
      </form>
    </Popover>
  )
}
