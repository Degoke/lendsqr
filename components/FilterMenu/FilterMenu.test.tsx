import { render, screen } from '@testing-library/react'
import FilterMenu from './index'

describe('FilterMenu tests', () => {
  it('renders Filtermenu component unchanged', () => {
    const { container } = render(<FilterMenu isOpen={false} />)
    expect(container).toMatchSnapshot()
  })

  it('renders Filtermenu when is open is true', () => {
    render(<FilterMenu isOpen={true} />)
    expect(screen.getByRole('form')).toBeInTheDocument()
  })

  it('dosent render Filtermenu when is open is false', () => {
    render(<FilterMenu isOpen={false} />)
    expect(screen.queryByRole('form')).toBeNull()
  })
})
