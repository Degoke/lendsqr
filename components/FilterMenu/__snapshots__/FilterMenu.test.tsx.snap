// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilterMenu tests renders Filtermenu component unchanged 1`] = `
<div>
  <div
    class="popover popover"
  />
</div>
`;
